grand_challenge_forge/partials/example-evaluation-method/example-evaluation-method{{cookiecutter._}}/evaluate.py.j2

```diff
@@ -18,21 +18,35 @@ Any container that shows the same behavior will do, this is purely an example of
 Happy programming!
 """
 import json
+import multiprocessing
+import os
 {% if cookiecutter.phase.algorithm_inputs | has_image -%}
 from glob import glob
 import SimpleITK
 {%- endif %}
 import random
-from multiprocessing import Pool
 from statistics import mean
 from pathlib import Path
 from pprint import pformat, pprint
+from concurrent.futures import ThreadPoolExecutor
 
 
 INPUT_DIRECTORY = Path("/input")
 OUTPUT_DIRECTORY = Path("/output")
 GROUND_TRUTH_DIRECTORY = Path("ground_truth")
 
+# The optimal number of workers ultimately depends on how many
+# resources each process would call upon.
+# To limit this, update the Dockerfile GRAND_CHALLENGE_MAX_WORKERS
+environ_cpu_limit = os.getenv("GRAND_CHALLENGE_MAX_WORKERS")
+cpu_count = multiprocessing.cpu_count()
+MAX_WORKERS = min(
+    [
+        int(environ_cpu_limit or cpu_count),
+        cpu_count,
+    ]
+)
```

Comment:
Maybe better to put this into a function so that it cannot be accidentally modified by consumers and doesn't litter the namespace with extra variables? Could also use a one-liner:

```python
MAX_WORKERS = min(os.environ.get("GRAND_CHALLENGE_MAX_WORKERS", math.inf), multiprocessing.cpu_count())
```